tests/test_compute.py

```diff
@@ -535,3 +540,53 @@ def test_compute_environments(client):
     for env in response.json:
         if env["priceMin"] == 0:
             assert env["id"] == "ocean-compute"
+
+
+@pytest.mark.integration
+def test_compute_paid_env(
+    client, publisher_wallet, consumer_wallet, paid_c2d_env, web3
+):
+    valid_until = get_future_valid_until()
+    deployer_wallet = Account.from_key(os.getenv("FACTORY_DEPLOYER_PRIVATE_KEY"))
+    fee_token = get_datatoken_contract(web3, get_ocean_token_address(web3))
+    fee_token.functions.transfer(consumer_wallet.address, to_wei(80)).transact(
+        {"from": deployer_wallet.address}
+    )
+
+    ddo, tx_id, alg_ddo, alg_tx_id = build_and_send_ddo_with_compute_service(
+        client,
+        publisher_wallet,
+        consumer_wallet,
+        False,
+        None,
+        c2d_address=paid_c2d_env["consumerAddress"],
+        valid_until=valid_until,
+        c2d_environment=paid_c2d_env["id"],
+        fee_token_args=(fee_token, to_wei(80)),
+    )
+    sa_compute = get_first_service_by_type(alg_ddo, ServiceType.ACCESS)
+    sa = get_first_service_by_type(ddo, ServiceType.COMPUTE)
+    nonce, signature = get_compute_signature(client, consumer_wallet, ddo.did)
+
+    # Start the compute job
+    payload = {
+        "dataset": {"documentId": ddo.did, "serviceId": sa.id, "transferTxId": tx_id},
+        "algorithm": {
+            "serviceId": sa_compute.id,
+            "documentId": alg_ddo.did,
+            "transferTxId": alg_tx_id,
+        },
+        "signature": signature,
+        "nonce": nonce,
+        "consumerAddress": consumer_wallet.address,
+        "environment": paid_c2d_env["id"],
+        "signature": signature,
+    }
+
+    # Start compute with valid signature
+    response = post_to_compute(client, payload)
+    assert response.status == "200 OK", f"start compute job failed: {response.data}"
```

Comment:
Can we have a test where this compute env is not paid? What error message is thrown in that case?

Reply:
Sure, I'll do it :)

Reply:
@mariacarmina The compute tests are generally integration tests which combine a startOrder with a compute start job. In this case, the startOrder will fail from contracts, which is to be expected. Our initialize endpoint only provides the necessary data for the user to make the startOrder call (through .py, .js or contracts). This means that an unpaid compute test wouldn't be a test of initialize, nor computeStart, but of internal functions from contracts/py/js. I can add it if you are adamant, but instead I added the following:

- a correction to showcase that initializeCompute gives amount=0 for unpaid envs
- a test to show that initializeCompute gives an expected amount for paid envs
- the end-to-end compute test was unmodified and uses the internal startOrder function, as previously

Also, please note that in tests_validation we have a test called `test_fee_amount_not_paid` which fills in the blanks between these two. Basically, it is now tested for paid envs that initializeCompute behaves ok and a combination of correct startOrder + compute job start also works ok. Testing the startOrder functionality itself, I believe it is not in our scope, but let me know what you think.